components/provisioner/internal/gardener/provisioner.go

```diff
@@ -64,8 +64,9 @@ func (g *GardenerProvisioner) ProvisionCluster(cluster model.Cluster, operationI
 	}
 
 	region := cluster.ClusterConfig.Region
+	purpose := *cluster.ClusterConfig.Purpose
```

Comment:
Can be nil, in which case this would crash. 

```suggestion
	purpose := ""
	if cluster.ClusterConfig.Purpose != nil {
		purpose = *cluster.ClusterConfig.Purpose
	}
```